### PR TITLE
Updating all the dependencies of BaseApp

### PR DESCRIPTION
**Pull request by SudoSu-bham**
runtime-version updated to 45
Perl updated
python v3.11 is now available
Also added a note in the [README.md](https://github.com/SudoSu-bham/org.sugarlabs.BaseApp/blob/branch/24.04/README.md) which ask the user to not use `--user`  flags if the flatpak  applications are installed system-wide.

### CODE REVIEW

**Comment by SudoSu-bham** [comment]
I had to force-push because my **Committer detail** was not correct. No files are changed only the commit hash changed.

File: README.md
Comment (tchx84) [inline README.md]: Please remove this comment, it's better to keep instructions consistent about the installation being used. What we should be doing here is to double check that we use `--user` everywhere.

File: README.md
Comment (tchx84) [inline README.md]: This new BaseApp should iam for GNOME 46 runtime, which is currently on beta, so let's use that instead...

```
flatpak --user remote-add --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
flatpak --user install flathub-beta org.gnome.Sdk//46beta
flatpak --user install flathub-beta org.gnome.Platform//46beta
```

File: README.md
Comment (tchx84) [inline README.md]: `46beta` instead

**Review by tchx84 — CHANGES_REQUESTED** [review]

**Comment by tchx84** [comment]
I ran this locally as well,  and is all good. This just needs to sit for the GNOME 46 runtime to be officially released

```
INFO    src.manifest: Loading module from shared-modules/dbus-glib/dbus-glib.json
INFO    src.manifest: Loading module from shared-modules/intltool/intltool-0.51.json
INFO    src.manifest: Checking 12 external data items
INFO    src.manifest: Skipped check [1/12] archive dbus-glib/dbus-glib-0.112.tar.gz (from shared-modules/dbus-glib/dbus-glib.json)
INFO    src.manifest: Skipped check [2/12] archive intltool/intltool-0.51.0.tar.gz (from shared-modules/intltool/intltool-0.51.json)
INFO    src.manifest: Started check [3/12] file python3-empy/empy-4.0.1.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [4/12] archive perl-XML-Simple/XML-Simple-2.25.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [5/12] archive xcursorgen/xcursorgen-xcursorgen-1.0.8.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [6/12] archive icon-naming-utils/icon-naming-utils-0.8.90.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [7/12] archive sugar-artwork/sugar-artwork-0.121.tar.xz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [8/12] archive python-dbus/dbus-python-1.3.2.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [9/12] file python3-decorator/decorator-5.1.1.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [10/12] git telepathy-glib/telepathy-glib.git (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [11/12] archive sugar-toolkit-gtk3/sugar-toolkit-gtk3-0.121.tar.xz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Started check [12/12] git sugarapp/sugarapp.git (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [3/12] file python3-empy/empy-4.0.1.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [4/12] archive python-dbus/dbus-python-1.3.2.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [5/12] file python3-decorator/decorator-5.1.1.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [6/12] git sugarapp/sugarapp.git (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [7/12] git telepathy-glib/telepathy-glib.git (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [8/12] archive sugar-artwork/sugar-artwork-0.121.tar.xz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [9/12] archive sugar-toolkit-gtk3/sugar-toolkit-gtk3-0.121.tar.xz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [10/12] archive perl-XML-Simple/XML-Simple-2.25.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [11/12] archive xcursorgen/xcursorgen-xcursorgen-1.0.8.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.manifest: Finished check [12/12] archive icon-naming-utils/icon-naming-utils-0.8.90.tar.gz (from org.sugarlabs.BaseApp.json)
INFO    src.main: Check finished with 0 error(s)

```

File: README.md
Comment (tchx84) [inline README.md]: Now that GNOME 46 is released, we should update all these references to the "46" branch, instead of "beta46".

**Review by tchx84 — CHANGES_REQUESTED** [review]

**Comment by tchx84** [comment]
@SudoSu-bham apologies for the delay, the only missing piece here is https://github.com/tchx84/sugarapp/issues/14 (otherwise all the appdata files would generate a deprecated author field)

**Comment by tchx84** [comment]
> @SudoSu-bham apologies for the delay, the only missing piece here is [tchx84/sugarapp#14](https://github.com/tchx84/sugarapp/issues/14) (otherwise all the appdata files would generate a deprecated author field)

@SudoSu-bham thanks for fixing that, I made a new sugarapp release, please update this PR (also, please double check everything is still up-to-date), so we can merge it.

File: README.md
Comment (tchx84) [inline README.md:10]: I don't think we want to remove the `-y` arg.

**Review by tchx84 — CHANGES_REQUESTED** [review]
Besides that one misc comment, all is missing is bumping the sugarapp release.

**Comment by SudoSu-bham** [comment]
> > @SudoSu-bham apologies for the delay, the only missing piece here is [tchx84/sugarapp#14](https://github.com/tchx84/sugarapp/issues/14) (otherwise all the appdata files would generate a deprecated author field)
> 
> @SudoSu-bham thanks for fixing that, I made a new sugarapp release, please update this PR (also, please double check everything is still up-to-date), so we can merge it.

Thankyou for the guidance and review, I will change the `README.md`.
However there is an issue while building BaseApp.
Updating manifest to use `sugarapp v1.13` works fine, but when `python3-empy` is updated to `v4.1` the `sugar-artwork` is generating an error during its build process

```
./gtk-widgets.css.em:426:29: error: ParseError: unknown markup sequence: `$((`; extension markup `@((...))` invoked with no installed extension
<root>:1:1: from this context
make[4]: *** [Makefile:466: gtk-widgets-72.css] Error 1
make[4]: Leaving directory '/run/build/sugar-artwork/gtk3/theme/3.20'
make[3]: *** [Makefile:359: install-am] Error 2
make[3]: Leaving directory '/run/build/sugar-artwork/gtk3/theme/3.20'
make[2]: *** [Makefile:379: install-recursive] Error 1
make[2]: Leaving directory '/run/build/sugar-artwork/gtk3/theme'
make[1]: *** [Makefile:371: install-recursive] Error 1
make[1]: Leaving directory '/run/build/sugar-artwork/gtk3'
make: *** [Makefile:419: install-recursive] Error 1
Error: module sugar-artwork: Child process exited with code 2
```

**Comment by tchx84** [comment]
> markup

We can investigate that later, 

> > > @SudoSu-bham apologies for the delay, the only missing piece here is [tchx84/sugarapp#14](https://github.com/tchx84/sugarapp/issues/14) (otherwise all the appdata files would generate a deprecated author field)
> > 
> > 
> > @SudoSu-bham thanks for fixing that, I made a new sugarapp release, please update this PR (also, please double check everything is still up-to-date), so we can merge it.
> 
> Thankyou for the guidance and review, I will change the `README.md`. However there is an issue while building BaseApp. Updating manifest to use `sugarapp v1.13` works fine, but when `python3-empy` is updated to `v4.1` the `sugar-artwork` is generating an error during its build process
> 
> ```
> ./gtk-widgets.css.em:426:29: error: ParseError: unknown markup sequence: `$((`; extension markup `@((...))` invoked with no installed extension
> <root>:1:1: from this context
> make[4]: *** [Makefile:466: gtk-widgets-72.css] Error 1
> make[4]: Leaving directory '/run/build/sugar-artwork/gtk3/theme/3.20'
> make[3]: *** [Makefile:359: install-am] Error 2
> make[3]: Leaving directory '/run/build/sugar-artwork/gtk3/theme/3.20'
> make[2]: *** [Makefile:379: install-recursive] Error 1
> make[2]: Leaving directory '/run/build/sugar-artwork/gtk3/theme'
> make[1]: *** [Makefile:371: install-recursive] Error 1
> make[1]: Leaving directory '/run/build/sugar-artwork/gtk3'
> make: *** [Makefile:419: install-recursive] Error 1
> Error: module sugar-artwork: Child process exited with code 2
> ```

I think we should keep the legacy version we have been using (same as fedora), so:

1. Revert the empy bump so we stay at  3.3.4.
2. Add the data checker to that module so that we only look for legacy releases, e.g, 3.3.x.

**Comment by SudoSu-bham** [comment]
Thankyou @tchx84 for selecting me this summer. I apologize for sending the PR late. 
I have downgraded the version of `python3-empy` to `3.3.4`. Additionally, I included a condition to ensure that the `flatpak external data checker` only looks for legacy versions. I tested both the `Physics` and `Pippy` activities, and they were functioning correctly with new `BaseApp`.

**Comment by tchx84** [comment]
> Thankyou @tchx84 for selecting me this summer. 

Glad to have you onboard :)

> I apologize for sending the PR late. I have downgraded the version of `python3-empy` to `3.3.4`. Additionally, I included a condition to ensure that the `flatpak external data checker` only looks for legacy versions. I tested both the `Physics` and `Pippy` activities, and they were functioning correctly with new `BaseApp`.

Excellent, Flathub is going through some hiccups today, so I will merge this after it's resolved. Then, we can continue updating everything else.

**Comment by tchx84** [comment]
@SudoSu-bham I unified some of your commits, since we don't need all the intermediate commits (e.g. like removing something and then adding it back within this same PR). Keep this in mind for the other upcoming PRs :)

Once this passes, I will merge this branch manually. And close this.

**Review by tchx84 — APPROVED** [review]

**Comment by tchx84** [comment]
Merged to https://github.com/flathub/org.sugarlabs.BaseApp/tree/branch/24.04 , now we wait until the branch is built and published.

**Comment by tchx84** [comment]
@SudoSu-bham `org.sugarlabs.BaseApp//24.04` is now publicly available, we can start updating the other PRs

**Comment by SudoSu-bham** [comment]
I have started the work, will start creating PR from today.

**Pull request closed without merging**
